src/goto-instrument/code_contracts.cpp

```diff
@@ -158,7 +158,7 @@ bool code_contractst::has_contract(const irep_idt fun_name)
          type.find(ID_C_spec_ensures).is_not_nil();
 }
 
-bool code_contractst::apply_contract(
+bool code_contractst::apply_function_contract(
```

Comment:
Comment applies to: d20bbe127aa03d8fd8d506c8c43f80871be850ce

Having this refactor take place in the same commit as other changes makes this harder to follow than necessary. This would be easier to follow if, for example, renamings happened in one commit (with an explanation for why the names are better unless obvious), one commit for replacing the loops with ranged-for (don't need an explanation for that beyond "ranged-for is better"), and then finally the actually functional change.

Reason for this is that refactorings should _not_ cause functional changes, whereas any bugfixes or features you add of course _should_ change the functionality, and it makes it a bit hard to see which is which if they're all combined in one changeset.

My preference would be if this commit was split up to that effect, although this is not a hard blocker. Worth considering for the future though!

Reply:
Thanks Hannes! This is a great suggestion. I just pushed this change.